### PR TITLE
Python 3: Port to Python3 for 10 pys

### PR DESCRIPTION
**Pull request by dzhengfy**
- Remote autotest reference
- Replace exceptions.Test* with test.*
- Other updates according to rules

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 1]
```diff
@@ -343,11 +344,11 @@ def create_mem_hotplug_xml(mem_size, mem_unit, numa_node='',
         mem_xml_file = os.path.join(data_dir.get_tmp_dir(),
                                     'memory_hotplug.xml')
         try:
-            fp = open(mem_xml_file, 'w')
-        except Exception, info:
+            with open(mem_xml_file, 'w') as fp:
+                fp.write(str(mem_xml))
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: with doesn't have an except clause

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 2]
```diff
@@ -548,8 +549,8 @@ def check_vm_state_dest():
             params["migrate_shared_storage"] = shared_storage = os.path.join(nfs_mount_path,
                                                                              (default_guest_asset))
 
-    for v in params.itervalues():
-        if isinstance(v, str) and v.count("EXAMPLE"):
+    for v in list(itervalues(params)):
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: use six module to deal with itervalues()

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_clone_wipe.py [hunk 3]
```diff
@@ -103,22 +101,22 @@ def run(test, params, env):
...
-            vol_name = utlv.get_vol_list(pool_name).keys()[0]
+            vol_name = list(utlv.get_vol_list(pool_name).keys())[0]
             logging.debug("Find partition %s in disk pool", vol_name)
         elif vol_format == 'sparse':
             # Create a sparse file in pool
             sparse_file = pool_target + '/' + vol_name
             cmd = "dd if=/dev/zero of=" + sparse_file
             cmd += " bs=1 count=0 seek=" + vol_capability
-            utils.run(cmd)
+            process.run(cmd, shell=True)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_clone_wipe.py]: maybe need add ignore_status=True parameter

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create_from.py [hunk 4]
```diff
@@ -114,25 +111,25 @@ def run(test, params, env):
...
             else:
-                raise error.TestFail("Expect fail, but run successfully!")
+                test.fail("Expect fail, but run successfully!")
     finally:
         # Cleanup: both src and dest should be removed
         try:
             pvt.cleanup_pool(src_pool_name, src_pool_type, src_pool_target,
                              src_emulated_image)
-        except error.TestFail, detail:
+        except test.fail as detail:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_create_from.py]: test.fail is one method,not exception object, so better throw exceptions:
```
from avocado.core import exceptions
exceptions.TestFail
```

**Comment by dzhengfy** [comment]
Local run with python 2:
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_capacity.dir_pool: PASS (50.31 s)
  (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_resize.positive_test.non_encrypt.non_acl.allocate_capacity.dir_pool: PASS (15.37 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_download_upload.download_upload.luks_encrypt.dir_pool.download.non_acl.0-1024: PASS (12.74 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_download_upload.download_upload.non_encrypt.dir_pool.download.non_acl.0-1024: PASS (13.85 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_download_upload.download_upload.non_encrypt.dir_pool.upload.non_acl.0-end: PASS (11.22 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.volume.dir_pool.vol_encrypt_none.vol_format_qcow2.vol_allocation.normal_size: PASS (8.67 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_create_from.positive_test.dest_vol_format.v_qcow2.src_vol_format.v_qcow2.dest_pool_type.dir.src_pool_type.dir: PASS (11.46 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_clone_wipe.positive_test.luks_encrypt.non_acl.vol_format.raw_f.pool_type.netfs: PASS (62.36 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_clone_wipe.positive_test.non_encrypt.non_acl.vol_format.qed_f.pool_type.dir: PASS (10.58 s)
(1/1) type_specific.io-github-autotest-libvirt.virsh.version.no_option: PASS (12.70 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vncdisplay.normal_test.name_option: PASS (9.23 s)

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: Remove the try except clause.

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create_from.py (hunk 4 above)
Comment (dzhengfy) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_create_from.py]: Done

**Pull request merged**